src/components/calculator.js

```diff
@@ -1,83 +1,133 @@
-import { Component } from 'react';
+import React, { useState } from 'react';
+import calculate from './logic/calculate';
 
-export default class Calculator extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
+const Calculator = () => {
+  const [display, newDisplay] = useState({ total: 0, next: null, sign: null });
 
-  render() {
-    return (
-      <div className="first">
-        <div className="calculator-wrap">
-          <div className="calculator-display">0</div>
-          <div className="calculator-keys">
-            <button type="button" className="all-clear" value="all-clear">
-              AC
-            </button>
-            <button type="button" className="operator" value="+/-">
-              +/-
-            </button>
-            <button type="button" className="operator" value="%">
-              %
-            </button>
-            <button type="button" className="operator" value="/">
-              &divide;
-            </button>
+  const buttonClicked = (event) => {
+    const btnName = event.target.innerText;
+    const answer = calculate(display, btnName);
+    newDisplay(answer);
+  };
+  const { total, next, sign } = display;
 
-            <button type="button" value="7">
-              7
-            </button>
-            <button type="button" value="8">
-              8
-            </button>
-            <button type="button" value="9">
-              9
-            </button>
-
-            <button type="button" className="operator" value="*">
-              &times;
-            </button>
-            <button type="button" value="4">
-              4
-            </button>
-            <button type="button" value="5">
-              5
-            </button>
-            <button type="button" value="6">
-              6
-            </button>
-
-            <button type="button" className="operator" value="-">
-              -
-            </button>
-            <button type="button" value="1">
-              1
-            </button>
-            <button type="button" value="2">
-              2
-            </button>
-
-            <button type="button" value="3">
-              3
-            </button>
-            <button type="button" className="operator" value="+">
-              +
-            </button>
-          </div>
-          <div className="last-three">
-            <button type="button" className="decimal" value="0">
-              0
-            </button>
-            <button type="button" value="decimal">
-              .
-            </button>
-            <button type="button" className="operator" value="=">
-              =
-            </button>
-          </div>
+  return (
+    <div className="first">
+      <div className="calculator-wrap">
+        <div className="calculator-display">
```

Comment:
- [OPTIONAL] Please it will be nice to improve the user experience of your app, the first value concatenats with the next value, which makes it not that user friendly. Please kindly fix that.